packages/ui5-application-writer/src/index.ts

```diff
@@ -115,7 +114,7 @@ async function enableTypescript(basePath: string, fs?: Editor): Promise<Editor>
     const ui5App = {
         app: manifest['sap.app'],
         ui5: {
-            typesVersion: getTypesVersion(manifest['sap.ui5']?.dependencies?.minUI5Version)
+            minUI5Version: manifest['sap.ui5']?.dependencies?.minUI5Version
```

Comment:
I am not sure I understand this change and the changes required down the line. If you just change this line back to what it was, with a minor fix : `typesVersion: getEsmTypesVersion(manifest['sap.ui5']?.dependencies?.minUI5Version)` then you don't need any changes in
packages/ui5-application-writer/src/options.ts and
packages/ui5-application-writer/templates/optional/typescript/package.json

Reply:
So there are a few things I've noticed here;
1. `enableTypescript` is an exposed method by the ui5-app-writer, used when adding TS to an existing Fiori app
2. It calls `enableTypescriptOption` which in turn calls`enableTypescript` from the factories flow which adds the actual templates
3. Your change will work for the `adding to an existing Fiori app` but the factories flow is used when TS is added as part of the options flow, used when creating the app for the first time
4. The optional flow, is passing the FeatureInput object as pass by reference thus if I overwrite the typesVersion in the factories `enableTypescript` method it will use that version also for the `@sapui5/ts-types` version.

One solution is to clone the FeatureInput object in the `enableTypescript` factories method but the minUI5Version still needs to be exposed.

Reply:
I need to try it myself with https://github.com/SAP/open-ux-tools/pull/751. The only problem that my proposal caused was in the test case where we enable `typescript` and `codeAssistant` in the same config. However, the test that my change broke was actually a problem.
Was your current code, the test results in the generation of a project with a types version of 1.71 and an esm types version of 1.90 which is not a good idea. Both should have the same version. If it is a ts project, then both should use the esm version. 
Please check https://github.com/SAP/open-ux-tools/pull/751 as a proposal to fix that inconsistency.